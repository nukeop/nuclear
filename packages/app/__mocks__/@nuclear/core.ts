const initialStoreState = () => ({
  equalizer: {
    selected: 'Default'
  },
  downloads: [],
  favorites: {
    albums: [],
    tracks: []
  },
  playlists: []
});

let mockStore = initialStoreState();

module.exports = {
  store: {
    get: (key: string) => mockStore[key] || {},
    set: (key: string, value: any) => {
      mockStore[key] = value;
    },
    clear: () => mockStore = initialStoreState()
  },
  createApi: () => ({
    app: {},
    store: {},
    React: jest.fn(),
    ReactDOM: jest.fn()
  }),
  getOption: () => '',
  rest: {
    LastFmApi: class {
      constructor() {}

      getTagInfo() {}
      getTagTracks() {}
      getTagAlbums() {}
      getTagArtists() {}
      getTopTags = () => Promise.resolve()
      getTopTracks = () => Promise.resolve()
    }
  },
  settingsConfig: jest.requireActual('@nuclear/core/src/settings').settingsConfig,
  SettingType: {
    BOOLEAN: 'boolean',
    LIST: 'list',
    NODE: 'node',
    NUMBER: 'number',
    STRING: 'string',
    DIRECTORY: 'directory'
  }
};
