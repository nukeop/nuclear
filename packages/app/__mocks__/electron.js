module.exports = {
  app: {
    getPath: () => { }
  },
  remote: {
    transformSource: () => { },
    app: {
      getPath: () => { }
    },
    dialog: {
      showOpenDialog: jest.fn(async () => Promise.resolve({
        canceled: false,
        filePaths: ['test file.txt']
      })),
      showSaveDialog: jest.fn(async () => Promise.resolve({
        canceled: false,
        filePath: 'downloaded_playlist'
      }))
    }
  },
  shell: {
    openExternal: jest.fn(async (link) => Promise.resolve({
      link
    }))
  }
};
