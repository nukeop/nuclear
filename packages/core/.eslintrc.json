{
  "extends": [
    "../../.eslintrc.json"
  ],
  "env": {
    "node": true,
    "browser": true,
    "commonjs": true
  },
  "rules": {
    "node/no-missing-import": 0,
    "node/no-extraneous-import": 0
  }
}
