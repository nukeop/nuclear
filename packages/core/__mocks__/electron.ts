module.exports = {
  app: {
    getPath: () => { }
  },
  remote: {
    transformSource: () => { },
    app: {
      getPath: () => { }
    },
    dialog: {
      showOpenDialog: jest.fn(async () => Promise.resolve({
        canceled: false,
        filePaths: ['test file.txt']
      }))
    }
  }
};
