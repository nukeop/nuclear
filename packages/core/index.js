export {
  LastFmApi,
  isElectron
} from './lib';
