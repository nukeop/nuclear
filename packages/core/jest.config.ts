export default async () => ({
  transform: {
    '^.+\\.(ts|tsx|js|jsx)$': 'ts-jest'
  },
  moduleNameMapper: {
    '^react$': '<rootDir>/../../node_modules/react'
  }
});
