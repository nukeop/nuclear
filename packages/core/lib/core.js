'use strict';

module.exports = core;

function core() {
  // TODO
}
