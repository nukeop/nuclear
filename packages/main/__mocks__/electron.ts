export const app = {
  on: jest.fn(),
  emit: jest.fn(),
  getPath: () => '',
  commandLine: {
    appendSwitch: jest.fn()
  },
  requestSingleInstanceLock: jest.fn(),
  quit: jest.fn()
};
export const TouchBar = {
  TouchBarButton: jest.fn(),
  TouchBarSlider: jest.fn()
};
export const nativeImage = {
  createFromPath: jest.fn().mockReturnValue({})
};
