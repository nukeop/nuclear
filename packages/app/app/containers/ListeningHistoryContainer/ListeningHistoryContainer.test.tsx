/* eslint-disable @typescript-eslint/ban-ts-comment */
import { waitFor } from '@testing-library/react';
import { store as electronStore } from '@nuclear/core';

import { buildElectronStoreState, buildStoreState } from '../../../test/storeBuilders';
import { AnyProps, mountedComponentFactory } from '../../../test/testUtils';
import { ipcRenderer } from 'electron';

jest.mock('electron-store', () => jest.fn().mockImplementation(() => ({
  get: jest.fn(),
  set: jest.fn()
})));

jest.mock('electron', () => ({
  ipcRenderer: {
    invoke: jest.fn().mockResolvedValue({
      data: [{
        uuid: 'test',
        artist: 'test artist',
        title: 'test title',
        createdAt: new Date('2020-01-01')
      }, {
        uuid: 'test2',
        artist: 'test artist2',
        title: 'test title 2',
        createdAt: new Date('2020-01-02')
      }]
    })
  }
}));

describe('Listening history container', () => {

  it('renders the listening history', () => {
    const { component } = mountComponent();

    waitFor(() => expect(component.getByText('test artist - test title')).toBeInTheDocument());
    waitFor(() => expect(component.getByText('test artist2 - test title 2')).toBeInTheDocument());
  });

  it('can refresh history', async () => {
    const { component } = mountComponent();

    component.getByTestId('refresh-history').click();

    expect(ipcRenderer.invoke).toHaveBeenCalledTimes(2);
  });

  it('can clear history', async () => {
    const { component } = mountComponent();

    component.getByText('Clear').click();

    expect(component.getByText('test artist - test title')).not.toBeInTheDocument();
  });

  const mountComponent = (electronStoreState?: AnyProps) => {
    // @ts-ignore
    electronStore.init({
      ...buildElectronStoreState(electronStoreState)
    });
    
    return mountedComponentFactory(
      ['/listening-history'],
      buildStoreState()
        .withConnectivity()
        .build()
    )();
  };
});
