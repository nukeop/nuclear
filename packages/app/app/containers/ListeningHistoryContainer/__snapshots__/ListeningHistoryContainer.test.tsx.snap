// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Listening history container renders the listening history 1`] = `
<DocumentFragment>
  <div
    class="main_layout_container"
  >
    <div
      class="listening_history_view"
    >
      <div
        class="header_container"
      >
        title
      </div>
    </div>
  </div>
</DocumentFragment>
`;
